pkg/push/api/api.go

```diff
@@ -561,28 +569,125 @@ func (s *PushService) UpdatePush(
 		}
 		return nil, dt.Err()
 	}
-	return &pushproto.UpdatePushResponse{}, nil
+
+	if updatedPushPb != nil {
+		// For security reasons we remove the service account from the API response
+		updatedPushPb.FcmServiceAccount = ""
+	}
+	return &pushproto.UpdatePushResponse{
+		Push: updatedPushPb,
+	}, nil
 }
 
-func (s *PushService) validateUpdatePushRequest(
+func (s *PushService) updatePushNoCommand(
 	ctx context.Context,
 	req *pushproto.UpdatePushRequest,
 	localizer locale.Localizer,
-) error {
-	if req.Id == "" {
-		dt, err := statusIDRequired.WithDetails(&errdetails.LocalizedMessage{
+	editor *eventproto.Editor,
+) (*pushproto.UpdatePushResponse, error) {
+	if err := s.validateUpdatePushRequestNoCommand(ctx, req, localizer); err != nil {
+		return nil, err
+	}
+	var updatedPushPb *pushproto.Push
+	var updatePushEvent *eventproto.Event
+	tx, err := s.mysqlClient.BeginTx(ctx)
+	if err != nil {
+		s.logger.Error(
+			"Failed to begin transaction",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
 			Locale:  localizer.GetLocale(),
-			Message: localizer.MustLocalizeWithTemplate(locale.RequiredFieldTemplate, "id"),
+			Message: localizer.MustLocalize(locale.InternalServerError),
 		})
 		if err != nil {
-			return statusInternal.Err()
+			return nil, statusInternal.Err()
 		}
-		return dt.Err()
+		return nil, dt.Err()
 	}
-	if s.isNoUpdatePushCommand(req) {
-		dt, err := statusNoCommand.WithDetails(&errdetails.LocalizedMessage{
+	err = s.mysqlClient.RunInTransaction(ctx, tx, func() error {
+		pushStorage := v2ps.NewPushStorage(tx)
+		push, err := pushStorage.GetPush(ctx, req.Id, req.EnvironmentNamespace)
+		if err != nil {
+			return err
+		}
+
+		updated, err := push.Update(req.Name, req.Tags)
+		if err != nil {
+			return err
+		}
+		updatePushEvent, err = domainevent.NewEvent(
+			editor,
+			eventproto.Event_PUSH,
+			push.Id,
+			eventproto.Event_PUSH_UPDATED,
+			&eventproto.PushTagsUpdatedEvent{
+				Tags: req.Tags,
+			},
+			req.EnvironmentNamespace,
+			updated,
+			push,
+		)
+		if err != nil {
+			return err
+		}
+		if err = s.publisher.Publish(ctx, updatePushEvent); err != nil {
+			return err
+		}
+		updatedPushPb = updated.Push
+
+		return pushStorage.UpdatePush(ctx, updated, req.EnvironmentNamespace)
+	})
+	if err != nil {
+		if err == v2ps.ErrPushNotFound || err == v2ps.ErrPushUnexpectedAffectedRows {
```

Comment:
When checking errors, please use `errors.If` instead. Same for other places.
```suggestion
		if errors.Is(err, v2ps.ErrPushNotFound) || if errors.Is(err, v2ps.ErrPushUnexpectedAffectedRows) {
```

Reply:
Done, same for other places